backend/app/adapter/facebook/identityprovider_integration_test.go

```diff
@@ -0,0 +1,149 @@
+// +build integration all
+
+package facebook
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/short-d/app/mdtest"
+)
+
+func TestIdentityProvider_GetAuthorizationURL(t *testing.T) {
+	t.Parallel()
+	httpRequest := mdtest.NewHTTPRequestFake(
+		func(req *http.Request) (response *http.Response, e error) {
+			return nil, nil
+		})
+	clientID := "id_12345"
+	clientSecret := "client_secret"
+	redirectURI := "http://localhost/oauth/facebook/sign-in/callback"
+	identityProvider := NewIdentityProvider(httpRequest, clientID, clientSecret, redirectURI)
+
+	urlResponse := identityProvider.GetAuthorizationURL()
+
+	parsedUrl, err := url.Parse(urlResponse)
+	mdtest.Equal(t, nil, err)
+	mdtest.Equal(t, "https", parsedUrl.Scheme)
+	mdtest.Equal(t, "www.facebook.com", parsedUrl.Host)
+	mdtest.Equal(t, "/v4.0/dialog/oauth", parsedUrl.Path)
+	mdtest.SameElements(t, [2]string{"public_profile", "email"}, strings.Split(parsedUrl.Query().Get("scope"), ","))
```

Comment:
Can we move `strings.Split(parsedUrl.Query().Get("scope"), ",")` into its own line as assign it to a variable?
Second question: Do we have to specify length of `2` here?

Reply:
addressed comment